src/typing_extensions.py

```diff
@@ -1986,7 +1986,7 @@ class _ConcatenateGenericAlias(list):
         __class__ = typing._GenericAlias
 
         def __init__(self, origin, args):
-            super().__init__(args)
+            list.__init__(self, args)
```

Comment:
```suggestion
            # Cannot use `super().__init__` here because of the `__class__` assignment
            # in the class body (https://github.com/python/typing_extensions/issues/661)
            list.__init__(self, args)
```